src/main/scala/mesosphere/marathon/core/launchqueue/impl/LaunchQueueActor.scala

```diff
@@ -0,0 +1,186 @@
+
+  override def receive: Receive = LoggingReceive {
+    Seq(
+      receiveHandlePurging,
+      receiveMessagesToSuspendedActor,
+      receiveHandleNormalCommands
+    ).reduce(_.orElse[Any, Unit](_))
+  }
+
+  /**
+    * Handles purging of an actor.
+    *
+    * Terminating an actor with a PoisonPill is not instantaneous. It will still process
+    * all prior messages. While waiting for the termination of the actor, we might
+    * receive further messages to the actor which would potentially lead to recreating it.
+    *
+    * Thus, we mark the actor as suspended and save all messages which would normally
+    * have been sent to this actor. When we receive confirmation of the actor's death (Terminated),
+    * we will replay these messages to ourselves with the correct sender.
+    */
+  private[this] def receiveHandlePurging: Receive = {
+    case Purge(appId) =>
+      launchers.get(appId) match {
+        case Some(actorRef) =>
+          val deferredMessages: Vector[DeferredMessage] =
+            suspendedLaunchersMessages(actorRef) :+ DeferredMessage(sender(), ConfirmPurge)
+          suspendedLaunchersMessages += actorRef -> deferredMessages
+          suspendedLauncherPathIds += appId
+          actorRef ! PoisonPill
+        case None => sender() ! (())
+      }
+
+    case ConfirmPurge => sender() ! (())
+
+    case Terminated(actorRef) =>
+      launcherRefs.get(actorRef) match {
+        case Some(pathId) =>
+          launcherRefs -= actorRef
+          launchers -= pathId
+
+          suspendedLaunchersMessages.get(actorRef) match {
+            case None =>
+              log.warning("Got unexpected terminated for app {}: {}", pathId, actorRef)
+            case Some(deferredMessages) =>
+              deferredMessages.foreach(msg => self.tell(msg.message, msg.sender))
+
+              suspendedLauncherPathIds -= pathId
+              suspendedLaunchersMessages -= actorRef
+          }
+        case None =>
+          log.warning("Don't know anything about terminated actor: {}", actorRef)
+      }
+  }
+
+  private[this] def receiveMessagesToSuspendedActor: Receive = {
+    case msg @ Count(appId) if suspendedLauncherPathIds(appId) =>
+      deferMessageToSuspendedActor(msg, appId)
+
+    case msg @ Add(app, count) if suspendedLauncherPathIds(app.id) =>
+      deferMessageToSuspendedActor(msg, app.id)
+
+    case msg @ RateLimiterActor.DelayUpdate(app, _) if suspendedLauncherPathIds(app.id) =>
+      deferMessageToSuspendedActor(msg, app.id)
+  }
+
+  private[this] def deferMessageToSuspendedActor(msg: Any, appId: PathId): Unit = {
+    val actorRef = launchers(appId)
+    val deferredMessages: Vector[DeferredMessage] =
+      suspendedLaunchersMessages(actorRef) :+ DeferredMessage(sender(), msg)
+    suspendedLaunchersMessages += actorRef -> deferredMessages
+  }
+
+  private[this] def receiveHandleNormalCommands: Receive = {
+    case List =>
+      import context.dispatcher
+      val scatter = launchers
```

Comment:
Here you do not filter for suspended launchers. See my other question on count, if that is correct.